yolox/utils/metric.py

```diff
@@ -24,6 +24,9 @@ def get_total_and_free_memory_in_Mb(cuda_device):
         "nvidia-smi --query-gpu=memory.total,memory.used --format=csv,nounits,noheader"
     )
     devices_info = devices_info_str.read().strip().split("\n")
+    if "CUDA_VISIBLE_DEVICES" in os.environ:
```

Comment:
It's better to pass cuda_device as args (or you could get cuda_device value while `cuda_device` is None). WDYT? 
Also could you please tell me which case make you send this PR?

Reply:
I think it's common to use `CUDA_VISIBLE_DEVICES` for gpu selection, as it done in detectron2. Even if cuda_device is given through args, setting `CUDA_VISIBLE_DEVICES` breaks this logic and leads to errors.
My variant is more robust solution.